Docs - supervisor spec links (#16988)
apache/druid: docs/ingestion/kinesis-ingestion.md

  ~ under the License.
  -->

When you enable the Amazon Kinesis indexing service, you can configure supervisors on the Overlord to manage the creation and lifetime of Kinesis indexing tasks.

The supervisor oversees the state of the indexing tasks to coordinate handoffs, manage failures, and ensure that scalability and replication requirements are maintained.

Kinesis indexing tasks read events using the Kinesis shard and sequence number mechanism to guarantee exactly-once ingestion. 

> This topic contains configuration information for the Amazon Kinesis indexing service supervisor for Apache Druid.

Reviewer: ```suggestion
This topic contains configuration information for the Kinesis indexing service supervisor for Apache Druid.
```

## Setup

To use the Kinesis indexing service, you must first load the `druid-kinesis-indexing-service` core extension on both the Overlord and the MiddleManager. See [Loading extensions](../configuration/extensions.md#loading-extensions) for more information.

Review [Known issues](#known-issues) before deploying the `druid-kinesis-indexing-service` extension to production.

## `spec`

> This section outlines the configuration properties that are specific to supervisor-led streaming ingestion for Amazon Kinesis.

Reviewer: I think the original version (without "supervisor-led streaming") was easier to read and comprehend.

Reviewer 2: if you must mention supervisor: "Streaming ingestion supervisors" is easier to read than  "supervisor-led streaming"

```
</details>

### `ioConfig`

> The `ioConfig` influences how data is read from a source system, such as Apache Kafka, Amazon S3, a mounted filesystem, or any other supported source system.

Reviewer: ```suggestion
The `ioConfig` influences how data is read from a source system, such as Amazon Kinesis, Amazon S3, a mounted filesystem, or any other supported source system.
```
This this topic is specific to Kinesis, we should use it in the examples list instead of Kafka.

> The `ioConfig` defines how Druid reads data from a source system, such as Amazon Kinesis, Amazon S3, a mounted filesystem, or any other supported source system.

Reviewer: ```suggestion
The `ioConfig` object defines how Druid reads data from a source system, such as Amazon Kinesis, Amazon S3, a mounted filesystem, or any other supported source system.
```

> The following table outlines the `ioConfig` configuration properties specific to Amazon Kinesis.

Reviewer: ```suggestion
The following table outlines the `ioConfig` configuration properties specific to Kinesis.
```

> For configuration properties shared across all supervisor-led ingestion, refer to the [supervisor `ioConfig`](supervisor.md#ioconfig).

Reviewer: Same comment about "streaming ingestion methods" vs "supervisor-led ingestion".

You can use `parser` to read [`thrift`](../development/extensions-contrib/thrift.md) formats.

### `tuningConfig`

> Specify tuning properties in a `tuningConfig` object at the top level of an ingestion spec.

Reviewer: ```suggestion
You specify tuning properties in a `tuningConfig` object at the top level of an ingestion spec.
```

> The following table outlines the `tuningConfig` configuration properties specific to Amazon Kinesis.

Reviewer: ```suggestion
The following table outlines the `tuningConfig` configuration properties specific to Kinesis.
```

You can use `parser` to read [`thrift`](../development/extensions-contrib/thrift.md) formats.

### `tuningConfig`

You specify tuning properties in a `tuningConfig` object at the top level of an ingestion spec.

The following table outlines the `tuningConfig` configuration properties specific to Kinesis.

For properties common to all streaming ingestion configurations, see [supervisor `tuningConfig`](supervisor.md#tuningconfig).

|Property|Type|Description|Required|Default|
|--------|----|-----------|--------|-------|

> |`skipSequenceNumberAvailabilityCheck`|Boolean|Whether to enable checking if the current sequence number is still available in a particular Kinesis shard. If `false`, the indexing task attempts to reset the current sequence number, depending on the value of `resetOffsetAutomatically`. For more information on the `resetOffsetAutomatically` property, see [Supervisor tuning configuration](supervisor.md#tuningConfig).|No|`false`|

Reviewer: ```suggestion
|`skipSequenceNumberAvailabilityCheck`|Boolean|Whether to enable checking if the current sequence number is still available in a particular Kinesis shard. If `false`, the indexing task attempts to reset the current sequence number, depending on the value of `resetOffsetAutomatically`. For more information on the `resetOffsetAutomatically` property, see [Supervisor tuning configuration](supervisor.md#tuningconfig).|No|`false`|
```
